packages/calypso-e2e/src/rest-api-client.ts

```diff
@@ -0,0 +1,91 @@
+import fetch, { RequestInit } from 'node-fetch';
+
+type RequestMethods = 'post' | 'get';
+type RequestParams = {
+	method: 'post' | 'get';
+	headers: GenericDict;
+	body?: GenericDict;
+};
+type GenericDict = { [ key: string ]: string | number };
+
+interface Site {
+	ID: string;
+	name: string;
+	description: string;
+	URL: string;
+	site_owner: string;
+}
+interface MySitesResponse {
+	sites: Site[];
+}
+
+const BASE_URL = 'https://public-api.wordpress.com/rest/v1.1/';
```

Comment:
As a heads up, the prefixes on this route can change depending on the API call and version! We have a v1.2 and v1.3, e.g.. Also, there's the .org rest API.

Maybe we should just have this be the host `https://public-api.wordpress.com`? And each route definition can handle specifying the needed version?